cou/commands.py

```diff
@@ -153,6 +153,22 @@ def get_subcommand_common_opts_parser() -> argparse.ArgumentParser:
         type=batch_size_arg,
         default=1000,
     )
+    subcommand_common_opts_parser.add_argument(
+        "--purge",
+        help="Delete data from shadow tables. before cloud upgrade.\n"
+        "Default to disable purge.",
+        action=argparse.BooleanOptionalAction,
+        default=False,
+    )
+    subcommand_common_opts_parser.add_argument(
+        "--purge-before",
+        dest="purge_before",
+        help="Specifying –before will delete data from all shadow tables "
+        "that is older than the date provided. "
+        "Date strings may be fuzzy, such as 'Oct 21 2015'. "
+        "Without before the step will delete all the data.",
+        type=str,
```

Comment:
Would be good to have a validation like [here](https://github.com/canonical/charmed-openstack-upgrader/pull/433#discussion_r1636449737)

Reply:
done